kore/src/Kore/Internal/SideCondition/SideCondition.hs

```diff
@@ -0,0 +1,56 @@
+{-|
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+-}
+
+module Kore.Internal.SideCondition.SideCondition
+    ( Representation
+    , fromText
+    ) where
+
+import Control.DeepSeq
+    ( NFData
+    )
+import Data.Hashable
+    ( Hashable
+    , hash
+    )
+import Data.Text
+    ( Text
+    )
+import qualified Generics.SOP as SOP
+    ( Generic
+    , HasDatatypeInfo
+    )
+import qualified GHC.Generics as GHC
+
+import Kore.Debug
+    ( Debug
+    , Diff (..)
+    )
+
+data Representation =
```

Comment:
We could also get this from

```.hs
newtype Representation = Representation { getRepresentation :: Hashed Text }
```

and most of the instances would come for free, except `Debug` and `Diff` (and we can define catch-all instances for those in `Debug`).